### ISSUE TITLE
Anyway to increase file limit from 200mb?

### ISSUE THREAD

**Issue by BRUGraphicDesign** [abdeladim-s/subsai#54]
Hello, I love the program it works amazingly well! I just want to find out, if is there an easy way to increase the file limit from 200 MB to say 1 GB or higher.

edit: I found out you can add the config.toml file to bypass the limit, but in doing so it pretty much kills my machine.

**Comment by abdeladim-s** [comment]
Yes @BRUGraphicDesign, the media file size is limited to 200 mb in the webui. 
I would suggest to use the command line interface for large files.

**Comment by bigbeka** [comment]
@abdeladim-s thank you for a great work you are doing here.

Any chance to increase the limit for the WebUI?

**Comment by abdeladim-s** [comment]
@bigbeka, welcome .. 'Glad you found it useful.
I think you can run the webui with --maxsize argument and provide the maximum size you want. 
Keep in mind though that you just cannot view a media file beyond 200 mb (this is a limitation of the underlying library used to view the file) otherwise, you can run the transcription and export the subtitles normally.

**Comment by bigbeka** [comment]
@abdeladim-s Sorry, perhaps I should have been more specific: I am running SubsAI in a Docker Container with `docker compose up -d`. 

Would I be able to pass through `subsai-webui --server.maxMessageSize 500` to the build process?

**Comment by abdeladim-s** [comment]
@bigbeka, yes I think you can, but not to the build, rather to the run command. 
Let me know if it didn't work.

**Comment by bigbeka** [comment]
@abdeladim-s no joy, here are the steps I took:

`docker-compose.yml` file:
`version: '3.9'
services:
    subsai-webui:
        image: subsai-subsai-webui
        volumes:
            - '/home:/media_files'
        ports:
            - '8501:8501'
        restart: unless-stopped
        command: subsai-webui --server.maxMessageSize 10000`


Output after uploading 2009MB .mp4 file:
`File "/opt/conda/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 565, in _run_script
    exec(code, module.__dict__)
File "/opt/conda/lib/python3.10/site-packages/subsai/webui.py", line 555, in <module>
    run()
File "/opt/conda/lib/python3.10/site-packages/subsai/webui.py", line 548, in run
    webui()
File "/opt/conda/lib/python3.10/site-packages/subsai/webui.py", line 318, in webui
    subs = _transcribe(file_path, stt_model_name, model_config)
File "/opt/conda/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 194, in wrapper
    return cached_func(*args, **kwargs)
File "/opt/conda/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 223, in __call__
    return self._get_or_create_cached_value(args, kwargs)
File "/opt/conda/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 248, in _get_or_create_cached_value
    return self._handle_cache_miss(cache, value_key, func_args, func_kwargs)
File "/opt/conda/lib/python3.10/site-packages/streamlit/runtime/caching/cache_utils.py", line 302, in _handle_cache_miss
    computed_value = self._info.func(*func_args, **func_kwargs)
File "/opt/conda/lib/python3.10/site-packages/subsai/webui.py", line 190, in _transcribe
    subs = subs_ai.transcribe(media_file=file_path, model=model)
File "/opt/conda/lib/python3.10/site-packages/subsai/main.py", line 115, in transcribe
    return stt_model.transcribe(media_file)
File "/opt/conda/lib/python3.10/site-packages/subsai/models/faster_whisper_model.py", line 255, in transcribe
    segments, info = self.model.transcribe(media_file, **self.transcribe_configs)
File "/opt/conda/lib/python3.10/site-packages/faster_whisper/transcribe.py", line 294, in transcribe
    audio = decode_audio(audio, sampling_rate=sampling_rate)
File "/opt/conda/lib/python3.10/site-packages/faster_whisper/audio.py", line 46, in decode_audio
    with av.open(input_file, mode="r", metadata_errors="ignore") as container:
File "av/container/core.pyx", line 401, in av.container.core.open
File "av/container/core.pyx", line 265, in av.container.core.Container.__cinit__
File "av/container/core.pyx", line 285, in av.container.core.Container.err_check
File "av/error.pyx", line 336, in av.error.err_check`

**Comment by bigbeka** [comment]
@abdeladim-s Yes, using the `docker compose up -d` with the command you suggested and bigger file size, I tried different models and resulted in the same error.

1. I pulled the latest changes
2. `docker compose build`
3. This resulted in GPU missing error, as I am running it on CPU: `docker compose run -p 8501:8501 -v /path/to/your/media_files/folder:/media_files subsai-webui --server.maxMessageSize 10000`
4. I commended out GPU passthrough, and still getting the same error.

I think the issue might lie here, smaller files are getting uploaded, bigger files are not getting uploaded:
<img width="317" alt="Screenshot 2024-03-11 at 09 56 18" src="https://github.com/abdeladim-s/subsai/assets/53529055/7aecdb20-55f2-483f-a577-f8b2575ff8ff">

I am running this in a server, which I need to make available to my team, for this use case, UI upload of bigger file sizes need to work for upload and transcription.

**Comment by abdeladim-s** [comment]
@bigbeka, Oh .. so you are running it on a server and you are using the upload option! 
To increase the limit in this case you'll need to use this command: `--server.maxUploadSize 10000` instead of `--server.maxMessageSize`, 
please try it as usual and let me know how it goes:
`docker compose run -p 8501:8501 -v /path/to/your/media_files/folder:/media_files subsai-webui --server.maxUploadSize 10000`

**Comment by abdeladim-s** [comment]
@bigbeka, 
Yes please if you can share your setup so I can understand the situation correctly, I indeed believe the `AxiosError` is related to the web host you are using or maybe you are using a Proxy of some sort?  

Can you re-upload the `subsai.log` file, because the link returns 404! 
And yeah I think we can move this to a new issue since the file limit problem is solved! 

----

If you are experiencing high RAM usage, I think you are still using `--server.maxMessageSize`, because otherwise I remember I fixed this issue by removing the video viewer if the file exceeds the limit, and you should get a warning message instead. Can you please check your docker compose file ?

**Comment by InternetRebelAKAQuachHungPhong** [comment]
So, is there a way to modify `maxMessageSize` within a file itself instead of having to feed a command argument, since I installed using `pipxu`?

**Comment by abdeladim-s** [comment]
> So, is there a way to modify `maxMessageSize` or `maxUploadSize` within a file itself instead of having to feed a command argument, since I installed using `pipxu`?

@InternetRebelAKAQuachHungPhong, What do you mean? You can pass arguments to packages installed with `pipxu` the same way you do with any other entry script! Isn't that the case?

**Comment by InternetRebelAKAQuachHungPhong** [comment]
Well then, where could that `config.toml` file the original 'comment' mentioned be?

Well, I'm talking about which file can contain `maxMessageSize` or `maxUploadSize` to modify so I can just launch with those settings by just entering `subsai-webui`.

**Comment by abdeladim-s** [comment]
Oh .. I got you. 
The `config.toml` file is located at
> ~/.streamlit/config.toml for macOS/Linux or %userprofile%/.streamlit/config.toml for Windows:

Add the following lines: 
```shell
[server]
maxUploadSize = 200
```

**Comment by InternetRebelAKAQuachHungPhong** [comment]
So, did you get the file or you created it? Because I didn't find it in my machine, then created it myself, and achieved the change I wanted. Guess this may be closed now.

**Comment by InternetRebelAKAQuachHungPhong** [comment]
> This looks like an issue with RAM, once the file upload finishes, RAM usage shoots up even before starting the transcription. Smaller (<300mb) files were fine using anything under 4GB RAM, for a file that is 1.78GB big, 12GB RAM was sufficient. But during transcription, only 6GB RAM is used. >

Oh no, now I wonder if there's a way to just slow down the processing when the maximum RAM is 'low', like 8-16 GB.

**Comment by abdeladim-s** [comment]
> Oh no, now I wonder if there's a way to just slow down the processing when the maximum RAM is 'low', like 8-16 GB. 

For bigger files, you can use the CLI instead.
